### DISCUSSION TITLE
Endpoint slack diagram vs. Timing Report

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#4578] [General]
This is just an observation that puzzled me for a bit that I thought I'd share.

First I thought something was wrong, but having thought about it for a bit, it's obvious: the endpoint slack diagram counts only endpoints, whereas the timing report has all the start and end point combinations. This means that the timing report has many more rows than there are endpoints in the slack diagram.

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/b44d56b6-1c12-45e7-a06f-f3238dbb420d)

At least 9999(maximum in GUI) rows here:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/dad6dfa1-8350-460e-9976-341d365f8d85)

Of course this is nothing new, there is a feature to display only one row per endpoint:

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/fec8e633-03df-4aaf-82f9-a67802050942)

Hmm... is that the worst slack per end-point?

**Comment by maliberty** [comment]
Yes it should be the worst slack path to that endpoint.  Total paths is exponential in design size.
